is-valid-turn.js

```diff
@@ -0,0 +1,89 @@
+"use strict";
+const isValidMove = require("./is-valid-move");
+const makeMove = require("./make-move");
+const _ = require("lodash");
+
+function isValidTurn(gameState, diceRoll, proposedMoves) {
+    function getAceyDeucey() {
+        const aceyDeucey = {};
+        diceRoll.forEach(function(roll, index) {
+            if (roll === 1 && !aceyDeucey.hasOwnProperty("oneLocation")) {
+                aceyDeucey.oneLocation = index; 
+            } else if (roll === 2 && !aceyDeucey.hasOwnProperty("twoLocation")) {
+                aceyDeucey.twoLocation = index;
+            } else if (!aceyDeucey.hasOwnProperty("doublesLocation")) {
+                aceyDeucey.doublesLocation = index;
+            } else {
+                aceyDeucey.isAceyDeucey = false;
+            }
+        });
+        return aceyDeucey;
+    }
+    
+    function proposedMovesMatchDice() {
+        if (diceRoll.length > 2) {
+                      
+            const aceyDeucey = getAceyDeucey();
+            if (aceyDeucey.isAceyDeucey === false) {
+                return false;
+            }
+
+            if (
+                !(proposedMoves[0].numberOfSpaces === 1 && proposedMoves[1].numberOfSpaces === 2) &&
+                !(proposedMoves[0].numberOfSpaces === 2 && proposedMoves[1].numberOfSpaces === 1)
+            ) {
+                return false;
+            }
+            
+            const sortedAceyDeuceyRoll = _.sortBy(diceRoll);
+            for (let i = 0; i < 3; i++) {
+                sortedAceyDeuceyRoll.push(diceRoll[aceyDeucey.doublesLocation]);
+            }
+            if (sortedAceyDeuceyRoll.length !== proposedMoves.length) {
+                return false;
+            }
+            return _(proposedMoves)
+                .map("numberOfSpaces")
+                .sort()
+                .every((numberOfSpaces, index) => numberOfSpaces === sortedAceyDeuceyRoll[index]);
+        }
+        
+        if (diceRoll[0] === diceRoll[1]) {
+            if (proposedMoves.length !== 4){
+                return false;
+            } 
+            return _(proposedMoves).map("numberOfSpaces").every(numberOfSpaces => numberOfSpaces === diceRoll[0]);
```

Comment:
You can also use `_.uniq` here.

```js
const uniqueNumbers = _(proposedMoves).map('numberOfSpaces').uniq().value();
```

Then, if you see that `uniqueNumbers.length === 1`, you'll know all the numbers in the input list were the same.